### PR TITLE
Feature/hw3

### PR DESCRIPTION
**Pull request by ValeriyPetryna**
_Fixed some issues relatively to previous HW2 reviewer comments_

**Project changes:**

0.  Removed bootstrap

1.   Added some material UI components (button, inputs, form fields, icon, errors)

2.   Changed project structure(added blog, shared header, and post form modules), (removed some extra layers, renamed some 
components)

**Styles and markup changes:**

1.   Fixed some styles, added variables

2.   Some minor semantic and structure changes in HTML elements, classes, attributes

**Core changes:**

1.   Used BehaviorSubject instead of Subject for Post publishing

2.   Minor changes (moved posts mock to the service layer, minor TS changes, etc.) 

3. Changed form validation logic

### CODE REVIEW

**Comment by MMaksymivUa** [comment]
Add screenshot

File: src/app/blog/components/blog-page/blog-page.component.scss [hunk 1]
```diff
@@ -0,0 +1,16 @@
+.content {
+  width: 80%;
+  margin: 0 auto;
+  display: flex;
+  justify-content: space-between;
+  padding: 10px;
+
+  &__blog {
+    width: 80%;
+    // flex-grow: 8;
```
Comment (MMaksymivUa) [inline src/app/blog/components/blog-page/blog-page.component.scss]: do not push commented code, it is a bad practice

File: src/app/blog/components/post/post.component.scss [hunk 2]
```diff
@@ -0,0 +1,65 @@
+@import '/src/styles/variables';
```
Comment (MMaksymivUa) [inline src/app/blog/components/post/post.component.scss:1]: define absolute path for this scss file. You can import them by using 
```
@import '_variables';
```

File: src/app/blog/components/post/post.component.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+})
+export class PostComponent implements OnInit {
+  @Input() post!: Post;
+
+  constructor() {}
+
+  ngOnInit(): void {}
+
+  likePost(): void {
+    this.post.likes = this.post.likes !== undefined ? this.post.likes + 1 : 0;
```
Comment (MMaksymivUa) [inline src/app/blog/components/post/post.component.ts]:
```
this.post?.likes = !this.post?.likes ? this.post.likes + 1 : 0
```

File: src/app/shared/components/post-form/post-form.component.html [hunk 4]
```diff
@@ -0,0 +1,47 @@
+<div data-target="#modal">
+  <form class="postForm" [formGroup]="postForm" (ngSubmit)="onSubmit()">
+    <div class="postForm__item">
+      <mat-form-field class="postForm__item-input" appearance="fill">
+        <mat-label>Author</mat-label>
+        <input
+          matInput
+          formControlName="author"
+          placeholder="John Snow"
+          required
```
Comment (MMaksymivUa) [inline src/app/shared/components/post-form/post-form.component.html]: you can pass it via formBuilder

**Comment by MMaksymivUa** [comment]
Good job !!!

**Review by MMaksymivUa — APPROVED** [review]

**Comment by ValeriyPetryna** [comment]
Added minor changes and screenshot
blog-page

**Pull request merged**